chap-cheri-riscv.tex

```diff
@@ -536,38 +476,26 @@ \subsubsection{Capability Exception Reporting}
 general-purpose capability register file.
 \end{description}
 
-\subsubsection{Capability Control and Status Registers (CCSRs)}
-\label{subsubsec-ccsrs}
-New per HART \xccsr{} \texttt{XLEN}-bit RISC-V CSRs are defined as per
-Figure~\ref{fig-ccsr} (shown for XLEN=32):
-
-\begin{figure}[!h]
-\begin{center}
-\begin{bytefield}[bitwidth=\textwidth/34]{32}
-  \bitheader[endianness=big]{0,1,2,31} \\
-  \bitbox{30}{\textbf{WIRI}}
-  \bitbox{1}{\texttt{d}}
-  \bitbox{1}{\texttt{e}}
-\end{bytefield}
-\caption{\xccsr{} register format; WIRI bits are Write Ignore Read Ignore.}
-\label{fig-ccsr}
-\end{center}
-\end{figure}
-
-\pmnote{WIRI has been removed from the RISC-V specification.  The spec
-  now typically uses WPRI (Write-Preserve, Read-Ignore) for such
-  undefined bits (e.g. in \texttt{mstatus}).  Both WARL (Write-Any,
-  Read-Legal) and WLRL (Write-Legal, Read-Legal) seem unsuitable here.
-  The RISC-V spec unsatisfyingly does not specify what happens if
-  software doesn't preserve values for WPRI, but then this spec
-  doesn't need to either.}
-
-\begin{description}
-\item [e] The \texttt{e} ``enable'' bit \ajnote{read only? / only wriFable in u-mode? only writable with a specific permission in pcc?} tells whether capability extensions are enabled or disabled.
-\item [d] The \texttt{d} ``dirty'' bit tells whether a capability register has been written. This is intended to help with memory requirements when implementing context switching.  \arnote{Is this still useful if we have hybrid and purecap co-process communication?}
-\hmnote{If the user-mode apps are purecap, and the kernel is hybrid (or does not do any cap ops at all), this bit might be useful
-to do lazy CHERI-context save/restore similar fo floating point.}
-\end{description}
+\subsubsection{CHERI Extension Control}
+
+For processors supporting the \texttt{Smstateen} extension, bit 2 of
+\mstateen{} and \sstateen{} is used to enable CHERI.  When CHERI is
+disabled, attempting to execute CHERI-specific instructions will raise
+an illegal instruction fault and existing instructions will always execute in
+integer encoding mode.
+
+\jhbnote{I'm not sure what to do about \PCC and \DDC when CHERI is
```

Comment:
IMO they continue to take full effect, and xEPCC vs xEPC should be done based on the target's state enabled bit, not the source's. That can include encoding mode too; ask for stupid things get stupid outcomes, and you can technically write code that runs in capability mode without using capability state (so long as you avoid AUIPCC), it's just a stupid thing to do. My main justification for allowing it is that it's simplifying for the architecture and microarchitecture and more consistent.